test/test.js

```diff
@@ -1,61 +1,210 @@
 import chai from 'chai'
+import sinon from 'sinon';
 import nock from 'nock'
 import es6promise from 'es6-promise'
 const expect = chai.expect
 
 es6promise.polyfill()
-import jsonFetch from '..'
+import jsonFetch, {retriers} from '..'
 
 describe('jsonFetch',() => {
-  it('resolves with json body for 200-level status codes', () => {
-    nock('http://www.test.com')
-      .get('/products/1234')
-      .reply(200, {name: 'apple'})
-    return jsonFetch('http://www.test.com/products/1234').then((response) => {
-      expect(response.body).to.deep.equal({name: 'apple'})
-      expect(response.status).to.equal(200)
-      expect(response.statusText).to.equal('OK')
-      expect(response.headers).to.be.ok
+  describe('single request with no retry', () => {
+    it('resolves with json body for 200-level status codes', () => {
+      nock('http://www.test.com')
+        .get('/products/1234')
+        .reply(200, {name: 'apple'})
+      return jsonFetch('http://www.test.com/products/1234').then((response) => {
+        expect(response.body).to.deep.equal({name: 'apple'})
+        expect(response.status).to.equal(200)
+        expect(response.statusText).to.equal('OK')
+        expect(response.headers).to.be.ok
+      })
     })
-  })
 
-  it('resolves with undefined for 404 status code', () => {
-    nock('http://www.test.com')
-      .get('/products/1234')
-      .reply(404, 'not found')
-    return jsonFetch('http://www.test.com/products/1234').then((response) => {
-      expect(response.body).to.equal(undefined)
-      expect(response.status).to.equal(404)
-      expect(response.statusText).to.equal('Not Found')
-      expect(response.headers).to.be.ok
+    it('resolves with undefined for 404 status code', () => {
```

Comment:
Recap of our inperson chat: @dannynelson proposal makes a lot of sense, but is a breaking change. He plans to make that change a separate PR to solicit feedback from the community. I'm going to merge this PR as is :-)